frontend/src/features/vessel_sidebar/beacon_malfunctions/VesselBeaconMalfunctions.tsx

```diff
@@ -15,12 +18,17 @@ export function VesselBeaconMalfunctions() {
   const { beaconMalfunctionsTab, loadingVesselBeaconMalfunctions, vesselBeaconMalfunctionsFromDate } = useAppSelector(
     state => state.beaconMalfunction
   )
-  const { selectedVessel } = useAppSelector(state => state.vessel)
+  const { selectedVesselIdentity } = useAppSelector(state => state.vessel)
+  const previousSelectedVesselIdentity = usePrevious(selectedVesselIdentity)
   const [isCurrentBeaconMalfunctionDetails, setIsCurrentBeaconMalfunctionDetails] = useState<boolean>(false)
 
   useEffect(() => {
     dispatch(getVesselBeaconMalfunctions(true) as any)
-  }, [dispatch, selectedVessel, vesselBeaconMalfunctionsFromDate])
+
+    if (!vesselsAreEquals(previousSelectedVesselIdentity, selectedVesselIdentity)) {
```

Comment:
Why not doing that inside the dispatcher via state selectors? (meaning `previousSelectedVesselIdentity` and `selectedVesselIdendity` would be in Redux state)

Reply:
We'll need to modify too much use-cases, if we move that to the dispatcher